File: External-Tutorial-SubComponentBuilder/src/main/java/zorail/rohan/com/subcomponent/module/AppModule.java
package zorail.rohan.com.subcomponent.module;

import android.app.Application;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;


import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by zorail on 11-May-17.
 */
@Module
public class AppModule {

    Application mApplication;
    public AppModule(Application mApplication)
    {
        this.mApplication = mApplication;
    }

    @Provides
    @Singleton
    Application provideApplication(){
        return mApplication;}
    @Provides
    @Singleton
    SharedPreferences provideSharedPreferences(Application application)
    {
        return PreferenceManager.getDefaultSharedPreferences(application);
    }


}

File: Tutorial8-1DaggerHilt-Basics/src/main/java/com/smarttoolfactory/tutorial8_1daggerhilt_basics/di/AppModule.kt
package com.smarttoolfactory.tutorial8_1daggerhilt_basics.di

import android.app.Application
import android.content.Context
import android.content.SharedPreferences
import com.smarttoolfactory.tutorial8_1daggerhilt_basics.model.ToastMaker
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ApplicationComponent
import dagger.hilt.android.qualifiers.ApplicationContext
import javax.inject.Singleton

@InstallIn(ApplicationComponent::class)
@Module
 class AppModule {

    @Singleton
    @Provides
    fun provideToastMaker(@ApplicationContext context: Context): ToastMaker {
        return ToastMaker(context)
    }

    @Provides
    @Singleton
    fun provideSharedPreferences( application: Application): SharedPreferences {
        return application.getSharedPreferences("PrefName", Context.MODE_PRIVATE)
    }
}